rdflib/namespace/__init__.py

```diff
@@ -228,6 +228,10 @@ def __contains__(cls, item):
             if issubclass(c, DefinedNamespace)
         )
 
+    def __dir__(cls) -> Iterable[str]:
```

Comment:
Technically this breaks the interface, as anything that depended on the default dir implementation would break. Not sure if this is a serious concern, but something we should think about.

```console
$ pipx run --spec rdflib==6.1.1 python3 -c 'from rdflib import RDF; print(dir(RDF))'
['_NS', '__annotations__', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_fail', '_underscore_num']
```

Other option is to provide a different method for this.

Reply:
> I highly doubt that this would be a problem for anyone though.

Yes, I highly doubt this will be a problem too and I'm in favour of reporposing the well-known `__dict__` for this, rather than adding a new method that people have to learn